release/create_github_release.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+# Copyright 2017 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# This script creates a release on github and uploads tar.gz/zip artifacts
+# for the release.  The release is created in draft/pre-release form, so
+# someone will likely need to visit the website to add release-notes and
+# then take the release out of draft/pre-release status.
+#
+# This script presumes that a tag for the release was previously created
+# on github (e.g., by running create_tag_reference.sh).
+#
+# This script relies on artifacts being in a local directory.  If you'd
+# instead like to specify a GCS source then consider running this script
+# via create_release.sh instead (don't forget to disable the other steps
+# in that script like releasing to gcs/gcr/docker, etc.).
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+ORG="istio"
+REPO="istio"
+TOKEN=""
+SHA=""
+VERSION=""
+REQUEST_FILE="$(mktemp /tmp/github.request.XXXX)"
+RESPONSE_FILE="$(mktemp /tmp/github.response.XXXX)"
+UPLOAD_DIR=""
+
+source ${SCRIPTPATH}/json_parse_shared.sh
+
+function usage() {
+  echo "$0
+    -k <file>  file that contains github user token
+    -o <name>  github org of repo (default is \"${ORG}\")
+    -r <name>  github repo name on which to create the release (default is \"${REPO}\")
+    -s <sha>   commit hash to use for release
+    -t <tok>   github user token to use in REST calls (use this or -k)
+    -u <dir>   source directory from which to upload artifacts (optional)
+    -v <ver>   version name for release"
+  exit 1
+}
+
+while getopts k:o:r:s:t:u:v: arg ; do
+  case "${arg}" in
+    k) TOKEN=$(< ${OPTARG});;
+    o) ORG="${OPTARG}";;
+    r) REPO="${OPTARG}";;
+    s) SHA="${OPTARG}";;
+    t) TOKEN="${OPTARG}";;
+    u) UPLOAD_DIR="${OPTARG}";;
+    v) VERSION="${OPTARG}";;
+    *) usage;;
+  esac
+done
+
+[[ -z "${ORG}" ]] && usage
+[[ -z "${REPO}" ]] && usage
+[[ -z "${TOKEN}" ]] && usage
+[[ -z "${SHA}" ]] && usage
+[[ -z "${VERSION}" ]] && usage
+
+echo "{\
```

Comment:
Prefer the following formula:
cat << EOF > ${REQUEST_FILE}
content
EOF

It is easier on the eyes.